### PR TITLE
Load window icons from a custom icon theme

### LINKED ISSUE #984: Window icons can be changed by the tileset
**Describe the bug**

Tilesets are able to override the window icon using `icon.freeciv` (and `icon.citydlg`). This shouldn't be allowed.

**To Reproduce**

1. In `misc/icons.spec` change `"icon.freeciv", "misc/civicon"` to `"icon.freeciv", "misc/cityicon"`.
2. Start the client.
3. Look at the window icon in the task bar.
4. Cry.

**Expected behavior**

The icon is part of the brand and cannot be overriden easily.

**Screenshots**

image
image

**Platform and version (please complete the following information):**
 - OS: all
 - Freeciv21 version: 3.0-beta1

**Additional context**

#978

### LINKED ISSUE #2019: The client cannot find its window icon
**Describe the bug**

```
[critical] freeciv21-client (.../freeciv21/client/tileset/tilespec.cpp:1208) - Could not load gfx file "misc/freeciv21-client".
```
and the window icon is missing.

**To Reproduce**
1. Start Freeciv21
2. Look at the icon in the task manager
3. A generic icon is used

**Expected behavior**

A shiny Freeciv21 icon is shown.

**Platform and version (please complete the following information):**
 - OS: Linux
 - Freeciv21 version: master

**Additional context**

The window icon doesn't work on Wayland for other reasons.
Solving #984 would also fix this issue.

### PR DESCRIPTION
**Pull request by lmoureaux**
The window icons used to be set by the tileset. This allowed user-made tilesets to override our branding and required loading tilesets before the first window was shown.

This commit moves the app icons away from tilesets into a custom icon theme (this should eventually be merged into the hicolor theme).

Closes #984.
Closes #2019.

Untested on Windows and Mac, but I tried to update all references to `misc/freeciv21-*.png`.

### CODE REVIEW

File: cmake/FreecivInstall.cmake [hunk 1]
```diff
@@ -169,6 +163,10 @@ if(UNIX AND NOT APPLE)
     message(STATUS "CMAKE_INSTALL_PREFIX changed from the default to /usr.")
   endif()
 
+  install(FILES ${CMAKE_SOURCE_DIR}/data/icons/index.theme
```
Comment (jwrober) [inline cmake/FreecivInstall.cmake:166]: Icon does not work on Windows. I think this install needs to happen on all platforms. If you install/copy the data/icons/* directory structure including the new file, then windows finds the icon.

**Review by jwrober — CHANGES_REQUESTED** [review]

File: data/CMakeLists.txt [hunk 2]
```diff
@@ -90,6 +90,33 @@ if (FREECIV_ENABLE_CLIENT OR FREECIV_ENABLE_CIVMANUAL)
     COMPONENT freeciv21)
 endif()
 
+# Icons
+if (FREECIV_ENABLE_CLIENT OR FREECIV_ENABLE_SERVER OR
+    FREECIV_ENABLE_RULEDIT OR FREECIV_ENABLE_FCMP_QT)
+  install(FILES ${CMAKE_SOURCE_DIR}/data/icons/index.theme
+          DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/freeciv21/icons
```
Comment (jwrober) [inline data/CMakeLists.txt:97]: DESTINATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}"/icons

as you have it, it installs to {root}/freeciv21/icons on msys2.

**Review by jwrober — COMMENTED** [review]
need another change

File: data/CMakeLists.txt (hunk 2 above)
Comment (jwrober) [reply data/CMakeLists.txt:97]: in the same file you can see the usage of the DESTINATION directive

Comment (lmoureaux) [reply data/CMakeLists.txt]: I hope I got it right this time :sweat_smile:

**Review by jwrober — APPROVED** [review]
Tada!

**Pull request merged**